analysis/rendering.py

```diff
@@ -21,8 +21,54 @@
 from common import utils
 
 
-def render_report(experiment_results, template, in_progress, coverage_report,
-                  description):
+def _warm_benchmark_plot_cache(benchmark_and_attrs):
+    """Warm up cache for benchmark plots by making plots. |benchmark_and_attrs|
+    is a tuple containing the benchmark and a list of attributes that when
+    accessed on the benchmark, cause plots to be created."""
+    benchmark, attrs = benchmark_and_attrs
+    for attr in attrs:
+        getattr(benchmark, attr)
+
+
+def _warm_cache(experiment_results, coverage_report, pool):
+    """"Warm plot caches by plotting plots for each benchmark in parallel before
+    it is done sequentially in jinja."""
+    # Plotting each benchmark's plots in parallel seems to be faster than
+    # plotting each plot in parallel. I suspect this is because each call to
+    # _warm_benchmark_plot_cache requires copying a benchmark, which has some
+    # overhead.
+    arguments = []
+    for benchmark in experiment_results.benchmarks:
+        attrs = []
+        if benchmark.type == 'bug':
+            attrs.extend([
+                'bug_box_plot',
```

Comment:
Could we somehow avoid having a redundant list of functions here, and figure them out automatically? One idea that comes to mind is to call the jinja render() with some fake/mock environment first which just collects the list of these functions.

Reply:
Umm... I agree having redundant functions is quite ugly. But I'm not sure this solution is better. I guess it is less likely to break. But it just feels a bit icky to use mock in production. Need to think about this more.